rllib/algorithms/impala/impala_learner.py

```diff
@@ -90,20 +90,12 @@ def compile_results(
 
 
 def _reduce_impala_results(results: List[ResultDict]) -> ResultDict:
-    """Reduce/Aggregate a list of results from Impala Learners.
-
-    Average the values of the result dicts. Add keys for the number of agent and env
-    steps trained.
+    """Reduce/Aggregate a list of results from Impala Learners by averaging.
 
     Args:
         results: result dicts to reduce.
 
     Returns:
         A reduced result dict.
     """
-    result = tree.map_structure(lambda *x: np.mean(x), *results)
-    agent_steps_trained = sum(r[ALL_MODULES][NUM_AGENT_STEPS_TRAINED] for r in results)
-    env_steps_trained = sum(r[ALL_MODULES][NUM_ENV_STEPS_TRAINED] for r in results)
-    result[ALL_MODULES][NUM_AGENT_STEPS_TRAINED] = agent_steps_trained
-    result[ALL_MODULES][NUM_ENV_STEPS_TRAINED] = env_steps_trained
-    return result
+    return tree.map_structure(lambda *x: np.mean(x), *results)
```

Comment:
what is different that we don't have to do this anymore? like sum of agent steps, etc?

Reply:
I was confusing it with the `Algorithm._counter` logic that we have to increase separately from this ALL_MODULES aggregation step.